src/DatePicker/Calendar.js

```diff
@@ -81,14 +83,19 @@ class Calendar extends Component {
         selectedDate: date,
       });
     }
+    if (nextProps.displayMode !== this.props.displayMode) {
```

Comment:
Got your point! What's about introducing two internal states like 'year_internal' and 'days_internal'?